src/serviceWorker/index.ts

```diff
@@ -19,31 +18,20 @@ self.skipWaiting()
 // This must be done before setting up workbox-precaching, so that it takes precedence.
 registerRoute(new DocumentRoute())
 
-// Splits entries into assets, which are loaded on-demand; and entries, which are precached.
-// Effectively, this precaches the document, and caches all other assets on-demand.
-const { assets, entries } = self.__WB_MANIFEST.reduce<{ assets: string[]; entries: PrecacheEntry[] }>(
-  ({ assets, entries }, entry) => {
-    if (typeof entry === 'string') {
-      return { entries, assets: [...assets, entry] }
-    } else if (entry.revision) {
-      return { entries: [...entries, entry], assets }
-    } else {
-      return { entries, assets: [...assets, toURL(entry)] }
-    }
-  },
-  { assets: [], entries: [] }
-)
+// Splits entries into:
+// - mediaURLs: loaded on-demand
+// - precacheEntries
+const { mediaURLs, precacheEntries } = groupEntries(self.__WB_MANIFEST)
```

Comment:
nit: i think the naming isn't really clear here - it seems like you basically just want to group URLs into either precache or on-demand (no cache), but i don't know why mediaURLs aren't cached as a reader of this. i think something like `mediaURLs` and `pageAssetURLs` or `ondemandEntries` and `precacheEntries` seems like a more clear grouping to me

Reply:
i had that before but zach felt differently about it. ill ask him again

Reply:
if he agrees i'll change in a diff pr.

Reply:
I like `onDemandEntries` and `precacheEntries`, but we are looking at the url of the on demand entries, so I think it simplifies the downstream code to use `{ onDemandURLs, precacheEntries }`.

Reply:
personally, think it'd be better to do `onDemandEntries` because then the function doesn't need to be coupled to the use of it. it's not complicated to them get the URLs from the list of onDemandEntries.

Reply:
OG cmment here https://github.com/Uniswap/interface/pull/6486#discussion_r1184016228